lib/ansible/module_utils/urls.py

```diff
@@ -218,9 +220,109 @@ def __init__(self, message, import_traceback, module=None):
         self.module = module
 
 
+class TLSinTLS:
+    def __init__(self, socket, context, server_hostname):
+        self.incoming = ssl.MemoryBIO()
+        self.outgoing = ssl.MemoryBIO()
+
+        self.socket = socket
+
+        self.sslobj = sslobj = context.wrap_bio(
+            self.incoming, self.outgoing, server_hostname=server_hostname
+        )
+
+        self.read = functools.partial(self._process, True, sslobj.read)
+        self.recv = functools.partial(self._process, True, sslobj.read)
+        self.send = functools.partial(self._process, False, sslobj.write)
+        self.sendall = functools.partial(self._process, False, sslobj.write)
+        self._decref_socketios = socket._decref_socketios
```

Comment:
Is something in the `http.client.HTTPSConnection` calling this internally?

Reply:
`socket.SocketIO` used in `makefile` uses it in the `close` method, which is called by `http`.  I could possibly get around this by not using `socket.SocketIO` directly, or subclass it.